autosar/package.py

```diff
@@ -386,9 +386,27 @@ def _createInternalBehavior(self, ws, swc, behaviorName, multipleInstance):
 
     def _createImplementation(self, swc, implementationName):
 
+        ws = self.rootWS()
+        assert(ws is not None)
         if implementationName is None:
             implementationName = swc.name+'_Implementation'
-        swc.implementation=autosar.component.SwcImplementation(implementationName,swc.behavior.ref,parent=self)
+        swc.implementation = autosar.component.SwcImplementation(implementationName, swc.behavior.ref, parent=self)
+
+        name = 'Default' if ws.version >= 4.0 else 'Code'

```

Comment:
Did you transform the code from component_writer.py to the lines below?
I cannot remember writing that strange code in component_writer.py which added a bunch of weird XML by default into the ScwImplementation. (I honestly have no idea what it even means.) 
To be sure I tried to create an empty Application component manually in DaVinci Developer and by default it just adds SHORT-NAME and BEHAVIOR-REF to the SWC-IMPLEMENTATION object and nothing else.

Do you see any reason why lines 395-408 should be executed by default? I'm thinking that perhaps these lines (395-408) can be executed as an option but not by default.

(If you feel uncomfortable doing the change I can do it myself after merge.)

Reply:
Yes, I did transform the code from component_writer.py 
The SWC unit tests is depending on that this information is filled in. So that is why I did move it from component_writer.py

I do not see the need of the lines at all, they can be removed, including line 389, 390.
But then all unit test that creates SWCs needs an update to remove this part in the expected arxml.

The code descriptor is used to describe the files included in the implementation of the SWC. For example do embedded coder generate a list of the files it generates for an SWC as a CodeDescriptor.
I have not seen any tools use this information when compiling the software.

Reply:
Let's merge this first then we can see what can be improved in a future commit.